pylint/testutils/_primer/primer_run_command.py

```diff
@@ -48,7 +47,13 @@ def run(self) -> None:
             "w",
             encoding="utf-8",
         ) as f:
-            json.dump(packages, f)
+            json.dump(
+                {
+                    p: [JSONReporter.serialize(m) for m in msgs]
```

Comment:
Is this serialisation even necessary? Aren't `TypedDict` dumpable and loadable by default?

Reply:
It's not a typed dict at this point, it's the actual message because when we load we deserialize.

Reply:
We could use the TypedDict we introduced in #7077 yes, but ``Message`` is used everywhere in the code base.

Reply:
And there an immutable ``location`` in it that I'd like to use later on.

Reply:
Ah, I missed the `JSONReporter` serialises all of its messages before printing anyway. That makes this necessary.